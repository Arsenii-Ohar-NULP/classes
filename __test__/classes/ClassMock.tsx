import Class from 'components/classes/Class';
import React from 'react';
export default function ClassMock({ cls }: { cls: Class }) {
  return <div data-testid={`class-${cls.id}`}>{cls.title}</div>;
}
