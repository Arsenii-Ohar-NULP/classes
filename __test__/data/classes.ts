import Class from 'components/classes/Class';

export const sampleFiveClasses: Class[] = [
    {
        id: 1,
        title: 'Class 1',
        description: 'This is class 1',
        teacher_id: 1,
        students_number: 1
    },
    {
        id: 2,
        title: 'Class 2',
        description: 'This is class 2',
        teacher_id: 2,
        students_number: 1
    },
    {
        id: 3,
        title: 'Class 3',
        description: 'This is class 3',
        teacher_id: 3,
        students_number: 1

    },
    {
        id: 4,
        title: 'Class 4',
        description: 'This is class 4',
        teacher_id: 4,
        students_number: 1
    },
    {
        id: 5,
        title: 'Class 5',
        description: 'This is class 5',
        teacher_id: 5,
        students_number: 1
    }
]