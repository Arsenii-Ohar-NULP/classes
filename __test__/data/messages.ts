import Message from 'pages/class/Message';

const sampleMessages : Message[] = [
    {
        id: 1,
        content: 'First',
        date: 'Yesterday',
        user: 1,
        cls: 1,
        fullname: 'Arsenii Ohar',
        username: 'arseniyohar'
    },
    {
        id: 2,
        content: 'First',
        date: 'Yesterday',
        user: 1,
        cls: 1,
        fullname: 'Arsenii Ohar',
        username: 'arseniyohar'
    },
    {
        id: 3,
        content: 'First',
        date: 'Yesterday',
        user: 1,
        cls: 1,
        fullname: 'Arsenii Ohar',
        username: 'arseniyohar'
    }

]