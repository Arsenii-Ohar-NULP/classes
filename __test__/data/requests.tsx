import { JoinRequest } from 'pages/class/JoinRequest'

export const getRequestsWithClassId = (classId: number, number: number) => {
    const requests: JoinRequest[] = []
    for (let i = 0; i < number; i++){
        requests.push(
            {
                userId: i + 1,
                classId: classId
            }
        )
    }

    return requests;
}