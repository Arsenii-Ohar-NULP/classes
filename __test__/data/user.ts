import User, { Role } from 'components/account/User';

export const sampleUser: User = {
    id: 1,
    username: 'seniorohar',
    firstName: 'Arsenii',
    lastName: 'Ohar',
    email: 'arsen.ogar@gmail.com',
    phone: '+380987669293',
    role: Role.Teacher   
}

export const sampleFiveUsers: User[] = [
    {
        id: 1,
        username: 'seniorohar',
        firstName: 'Arsenii',
        lastName: 'Ohar',
        email: 'arsen.ogar@gmail.com',
        phone: '+380987669293',
        role: Role.Student
    },
    {
        id: 2,
    username: 'vitalikpikh',
    firstName: 'Vitalik',
    lastName: 'Pikhotskiy',
    email: 'vitaliipik@gmail.com',
    phone: '+380967629253',
    role: Role.Student
    },
    {
        id: 3,
    username: 'svyatoslavshainoha',
    firstName: 'Svyatoslav',
    lastName: 'Shainoha',
    email: 'vitaliipik@gmail.com',
    phone: '+380967629253',
    role: Role.Student
    },
      {
        id: 4,
    username: 'olehrysin',
    firstName: 'Oleh',
    lastName: 'Rysin',
    email: 'olehrysin@gmail.com',
    phone: '+380967323273',
    role: Role.Student
    },
    {
        id: 5,
    username: 'johnadams',
    firstName: 'John',
    lastName: 'Adams',
    email: 'johnadams@gmail.com',
    phone: '+380967723273',
    role: Role.Student
    }
]