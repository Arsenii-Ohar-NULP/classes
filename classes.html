<!DOCTYPE html>
<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]>      <html class="no-js"> <!--<![endif]-->
<html>
  <head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title></title>
    <meta name="description" content="" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="classes.css" />
  </head>
  <body>
    <!--[if lt IE 7]>
      <p class="browsehappy">
        You are using an <strong>outdated</strong> browser. Please
        <a href="#">upgrade your browser</a> to improve your experience.
      </p>
    <![endif]-->
    <nav class="navbar bg-dark sticky-top">
      <div class="container-fluid justify-content-start">
        <a class="navbar-brand text-white" href="#"> Classes </a>
        <a class="nav-link active text-white" aria-current="page" href="#"
          >Home</a
        >
      </div>
    </nav>
    <div class="card m-3 text-center shadow">
      <h4 class="p-3">Courses we have for you</h4>
      <div class="d-flex p-3 flex-wrap justify-content-center">
        <div class="col-sm text-center">
          <img
            src="images/react.png"
            width="215"
            height="112"
            class="rounded"
          />
          <div class="vstack align-middle">
            <p class="fs-5 m-0 p-1 py-1 text-wrap">Learn React.js in a month</p>
            <p class="fs-6 px-1 m-0">Arsenii Ohar</p>
          </div>
        </div>
        <div class="col-sm m-1 pointer">
          <div class="text-center">
            <img
              src="images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
        <div class="col-md m-1 pointer">
          <div class="text-center">
            <img
              src="images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
        <div class="col-lg m-1 pointer">
          <div class="text-center">
            <img
              src="images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
        <div class="col-lg m-1 pointer">
          <div class="text-center">
            <img
              src="images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
      </div>
    </div>
    <div class="card m-3 text-center shadow">
      <h4 class="p-3">Your current courses</h4>
      <div class="d-flex p-3 flex-wrap justify-content-center">
        <div class="col-md text-center m-1">
          <img
            src="images/react.png"
            width="215"
            height="112"
            class="rounded"
          />
          <div
            class="progress h-2 w-auto m-2"
            role="progressbar"
            aria-label="Basic example"
            aria-valuenow="50"
            aria-valuemin="0"
            aria-valuemax="100"
          >
            <div class="progress-bar w-25 h-2"></div>
          </div>
          <div class="vstack align-middle">
            <p class="fs-5 m-0 p-1 py-1 text-wrap">Learn React.js in a month</p>
            <p class="fs-6 px-1 m-0">Arsenii Ohar</p>
          </div>
        </div>
        <div class="col-md m-1 pointer m-1">
          <div class="text-center">
            <img
              src="images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <div
            class="progress h-2 w-auto mx-auto m-2"
            role="progressbar"
            aria-label="Basic example"
            aria-valuenow="50"
            aria-valuemin="0"
            aria-valuemax="100"
          >
            <div class="progress-bar w-50 h-2"></div>
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
        <div class="col-md m-1 pointer m-1">
          <div class="text-center">
            <img
              src="images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <div
            class="progress h-2 w-auto mx-auto m-2"
            role="progressbar"
            aria-label="Basic example"
            aria-valuenow="50"
            aria-valuemin="0"
            aria-valuemax="100"
          >
            <div class="progress-bar w-50 h-2"></div>
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
      </div>
    </div>

    <script src="./node_modules/bootstrap/dist/js/bootstrap.bundle.js"></script>
  </body>
</html>
