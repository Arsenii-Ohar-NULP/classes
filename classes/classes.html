<!DOCTYPE html>
<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]>      <html class="no-js"> <!--<![endif]-->
<html>
  <head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title>Classes - home</title>
    <meta name="description" content="" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="classes.css" />
  </head>
  <body>
    <!--[if lt IE 7]>
      <p class="browsehappy">
        You are using an <strong>outdated</strong> browser. Please
        <a href="#">upgrade your browser</a> to improve your experience.
      </p>
    <![endif]-->
    <nav class="navbar navbar-expand-lg bg-body-tertiary" data-bs-theme="dark">
      <div class="container-fluid">
        <a class="navbar-brand" href="#">Classes</a>
        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNavDropdown"
          aria-controls="navbarNavDropdown"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div
          class="collapse navbar-collapse justify-content-between"
          id="navbarNavDropdown"
        >
          <ul class="navbar-nav">
            <li class="nav-item">
              <a class="nav-link active" aria-current="page" href="#">Home</a>
            </li>
          </ul>
          <ul class="navbar-nav">
            <li class="nav-item">
              <a class="nav-link" href="#">
                <img
                id="profilePic"
                  height="48"
                  width="48"
                  class="rounded-circle"
                />
              </a>
            </li>
            <li class="nav-item justify-content-center m-2">
              <button type="button" class="btn btn-primary" id="logoutButton">
                Logout
              </button>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div>
      <div class="text-center align-items-center" id="spinner">
        <div class="spinner-border text-primary" role="status">
          <span class="visually-hidden">Loading...</span>
        </div>
      </div>
    </div> 
    <div class="card m-3 text-center shadow visually-hidden" id="classesForYouDiv">
      <h4 class="p-3">Courses we have for you</h4>
      <div class="d-flex p-3 flex-wrap justify-content-center" id="coursesForYou">
        <!-- <div class="col-sm text-center">
          <img
            src="../images/react.png"
            width="215"
            height="112"
            class="rounded"
          />
          <div class="vstack align-middle">
            <p class="fs-5 m-0 p-1 py-1 text-wrap">Learn React.js in a month</p>
            <p class="fs-6 px-1 m-0">Arsenii Ohar</p>
          </div>
        </div>
        <div class="col-sm m-1 pointer">
          <div class="text-center">
            <img
              src="../images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
        <div class="col-md m-1 pointer">
          <div class="text-center">
            <img
              src="../images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
        <div class="col-lg m-1 pointer">
          <div class="text-center">
            <img
              src="../images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
        <div class="col-lg m-1 pointer">
          <div class="text-center">
            <img
              src="../images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
      </div> -->
      </div>
    </div>
    <div class="card m-3 text-center shadow visually-hidden" id="currentClassesDiv">
      <h4 class="p-3">Your current courses</h4>
      <div class="d-flex p-3 flex-wrap justify-content-center" id="currentClasses">
        <div class="visually-hidden" id="noClassesAssignedDiv">
          <div>
            There are no assigned classes for you. Choose a first one!
          </div>
        </div>
        <!-- <div class="col-md text-center m-1">
          <img
            src="../images/react.png"
            width="215"
            height="112"
            class="rounded"
          />
          <div
            class="progress h-2 w-auto m-2"
            role="progressbar"
            aria-label="Basic example"
            aria-valuenow="50"
            aria-valuemin="0"
            aria-valuemax="100"
          >
            <div class="progress-bar w-25 h-2"></div>
          </div>
          <div class="vstack align-middle">
            <p class="fs-5 m-0 p-1 py-1 text-wrap">Learn React.js in a month</p>
            <p class="fs-6 px-1 m-0">Arsenii Ohar</p>
          </div>
        </div>
        <div class="col-md m-1 pointer m-1">
          <div class="text-center">
            <img
              src="../images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <div
            class="progress h-2 w-auto mx-auto m-2"
            role="progressbar"
            aria-label="Basic example"
            aria-valuenow="50"
            aria-valuemin="0"
            aria-valuemax="100"
          >
            <div class="progress-bar w-50 h-2"></div>
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div>
        <div class="col-md m-1 pointer m-1">
          <div class="text-center">
            <img
              src="../images/maxresdefault.jpg"
              width="215"
              height="112"
              class="rounded"
            />
          </div>
          <div
            class="progress h-2 w-auto mx-auto m-2"
            role="progressbar"
            aria-label="Basic example"
            aria-valuenow="50"
            aria-valuemin="0"
            aria-valuemax="100"
          >
            <div class="progress-bar w-50 h-2"></div>
          </div>
          <p class="fs-5 m-0 p-1 py-1 text-wrap">
            Web with .NET Core and React.js
          </p>
          <p class="fs-6 m-0 px-1">Svyatoslav Shainoha</p>
        </div> -->
      </div>
    </div>

    <script type="module" src="../node_modules/bootstrap/dist/js/bootstrap.bundle.js"></script>
    <script type="module" src="./classLoader.js"></script>
  </body>
</html>
