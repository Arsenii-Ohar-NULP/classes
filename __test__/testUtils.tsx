import React, { PropsWithChildren } from 'react';
import { render, renderHook } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import type { RootState, AppStore } from 'pages/redux/store';
// As a basic setup, import your same slice reducers
import { authInitialState, authSlice } from 'pages/redux/auth';
import { classSlice, classesInitialState } from 'pages/redux/classes';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = { auth: authInitialState, classes: classesInitialState },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { auth: authSlice.reducer, classes: classSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderHookWithProviders(
  ui,
  {
    preloadedState = { auth: authInitialState, classes: classesInitialState },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { auth: authSlice.reducer, classes: classSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...renderHook(() => ui(), { wrapper: Wrapper, ...renderOptions }) };
}
