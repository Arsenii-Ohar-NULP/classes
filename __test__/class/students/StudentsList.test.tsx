import React from 'react';
import {sampleFiveClasses} from "../../data/classes";
import {renderWithProviders} from "../../testUtils";
import {StudentsList} from "../../../components/class/students/StudentsList";
import {fireEvent, screen, waitFor} from "@testing-library/react";
import {sampleFiveUsers } from "../../data/user";
import {act} from "react-dom/test-utils";

const navigateMock = jest.fn();
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    pathname: '/',
  }),
}));

describe('Students List', () => {
    it('should match a snapshot, when students are loaded', () => {
         const students = sampleFiveUsers;
        const isLoading = false;
        const cls = sampleFiveClasses[0];
        const onDelete = jest.fn();

        const {container} = renderWithProviders(<StudentsList isLoading={isLoading} students={students} cls={cls.id} onDelete={onDelete}/>);

        expect(container).toMatchSnapshot();
    })

    it('should return nothing, when students are loading(being fetched)', async () => {
        const isLoading = true;
        const students = null;
        const cls = sampleFiveClasses[0];
        const onDelete = null;

        const {container} = renderWithProviders(<StudentsList isLoading={isLoading}
                                          students={students}
                                          cls={cls.id}
                                          onDelete={onDelete}/> );



        await waitFor(() => {
            expect(container).toBeEmptyDOMElement();
        })
    });

    it('should return table with students info, when students are loaded', async () => {
        const students = sampleFiveUsers;
        const isLoading = false;
        const cls = sampleFiveClasses[0];
        const onDelete = jest.fn();

        renderWithProviders(<StudentsList isLoading={isLoading} students={students} cls={cls.id} onDelete={onDelete}/>);
        await waitFor(() => {
            for (const student of students){
                screen.findByText(`${student.firstName} ${student.lastName}`);
            }
        })
    })

    it('should show modal, when delete button of a student card is clicked', async () => {
        const students = sampleFiveUsers.slice(0, 1);
        const isLoading = false;
        const cls = sampleFiveClasses[0];
        const onDelete = jest.fn();

        renderWithProviders(<StudentsList isLoading={isLoading} students={students} cls={cls.id} onDelete={onDelete}/>);
        await act(async () =>  {
            const button = await screen.findByRole('button');
            fireEvent.click(button);
        })

        await waitFor(async () => {
            await screen.findByRole('dialog');
        })
    })
})

