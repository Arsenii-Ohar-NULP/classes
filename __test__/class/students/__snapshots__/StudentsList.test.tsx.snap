// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Students List should match a snapshot, when students are loaded 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="overflow-scroll hide-scrollbar smooth-shadow"
    >
      <table
        class="table shadow m-0"
      >
        <thead
          class="border px-2"
        >
          <tr>
            <th
              class="text-uppercase align-middle"
              scope="col"
            >
              <div
                class="ms-3"
              >
                Student
              </div>
            </th>
            <th
              class="text-uppercase align-middle"
              scope="col"
            >
              Email
            </th>
            <th
              class="text-uppercase align-middle"
              scope="col"
            >
              Phone Number
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="p-2 w-100 text-start overflow-scroll neat-text student"
            data-testid="student-card"
          >
            <td
              class="d-flex text-center gap-2 align-items-center"
              scope="row"
            >
              <img
                alt="Profile picture for seniorohar"
                class="rounded-circle bg-light border border-primary mx-2 "
                height="48"
                id="profilePic"
                src="https://api.dicebear.com/6.x/lorelei/svg/seed=seniorohar"
                width="48"
              />
              <div>
                Arsenii
                 
                Ohar
              </div>
            </td>
            <td
              scope="row"
            >
              arsen.ogar@gmail.com
            </td>
            <td
              scope="row"
            >
              <div
                class="d-flex gap-3"
              >
                +380987669293
                <div
                  class="invisible"
                  data-testid="invisible-button"
                >
                  <button
                    class="p-1 btn btn-danger"
                    type="button"
                  >
                    <img
                      alt="Delete Icon"
                      data-nimg="1"
                      decoding="async"
                      height="24"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                      style="color: transparent;"
                      width="24"
                    />
                  </button>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="p-2 w-100 text-start overflow-scroll neat-text student"
            data-testid="student-card"
          >
            <td
              class="d-flex text-center gap-2 align-items-center"
              scope="row"
            >
              <img
                alt="Profile picture for vitalikpikh"
                class="rounded-circle bg-light border border-primary mx-2 "
                height="48"
                id="profilePic"
                src="https://api.dicebear.com/6.x/lorelei/svg/seed=vitalikpikh"
                width="48"
              />
              <div>
                Vitalik
                 
                Pikhotskiy
              </div>
            </td>
            <td
              scope="row"
            >
              vitaliipik@gmail.com
            </td>
            <td
              scope="row"
            >
              <div
                class="d-flex gap-3"
              >
                +380967629253
                <div
                  class="invisible"
                  data-testid="invisible-button"
                >
                  <button
                    class="p-1 btn btn-danger"
                    type="button"
                  >
                    <img
                      alt="Delete Icon"
                      data-nimg="1"
                      decoding="async"
                      height="24"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                      style="color: transparent;"
                      width="24"
                    />
                  </button>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="p-2 w-100 text-start overflow-scroll neat-text student"
            data-testid="student-card"
          >
            <td
              class="d-flex text-center gap-2 align-items-center"
              scope="row"
            >
              <img
                alt="Profile picture for svyatoslavshainoha"
                class="rounded-circle bg-light border border-primary mx-2 "
                height="48"
                id="profilePic"
                src="https://api.dicebear.com/6.x/lorelei/svg/seed=svyatoslavshainoha"
                width="48"
              />
              <div>
                Svyatoslav
                 
                Shainoha
              </div>
            </td>
            <td
              scope="row"
            >
              vitaliipik@gmail.com
            </td>
            <td
              scope="row"
            >
              <div
                class="d-flex gap-3"
              >
                +380967629253
                <div
                  class="invisible"
                  data-testid="invisible-button"
                >
                  <button
                    class="p-1 btn btn-danger"
                    type="button"
                  >
                    <img
                      alt="Delete Icon"
                      data-nimg="1"
                      decoding="async"
                      height="24"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                      style="color: transparent;"
                      width="24"
                    />
                  </button>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="p-2 w-100 text-start overflow-scroll neat-text student"
            data-testid="student-card"
          >
            <td
              class="d-flex text-center gap-2 align-items-center"
              scope="row"
            >
              <img
                alt="Profile picture for olehrysin"
                class="rounded-circle bg-light border border-primary mx-2 "
                height="48"
                id="profilePic"
                src="https://api.dicebear.com/6.x/lorelei/svg/seed=olehrysin"
                width="48"
              />
              <div>
                Oleh
                 
                Rysin
              </div>
            </td>
            <td
              scope="row"
            >
              olehrysin@gmail.com
            </td>
            <td
              scope="row"
            >
              <div
                class="d-flex gap-3"
              >
                +380967323273
                <div
                  class="invisible"
                  data-testid="invisible-button"
                >
                  <button
                    class="p-1 btn btn-danger"
                    type="button"
                  >
                    <img
                      alt="Delete Icon"
                      data-nimg="1"
                      decoding="async"
                      height="24"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                      style="color: transparent;"
                      width="24"
                    />
                  </button>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="p-2 w-100 text-start overflow-scroll neat-text student"
            data-testid="student-card"
          >
            <td
              class="d-flex text-center gap-2 align-items-center"
              scope="row"
            >
              <img
                alt="Profile picture for johnadams"
                class="rounded-circle bg-light border border-primary mx-2 "
                height="48"
                id="profilePic"
                src="https://api.dicebear.com/6.x/lorelei/svg/seed=johnadams"
                width="48"
              />
              <div>
                John
                 
                Adams
              </div>
            </td>
            <td
              scope="row"
            >
              johnadams@gmail.com
            </td>
            <td
              scope="row"
            >
              <div
                class="d-flex gap-3"
              >
                +380967723273
                <div
                  class="invisible"
                  data-testid="invisible-button"
                >
                  <button
                    class="p-1 btn btn-danger"
                    type="button"
                  >
                    <img
                      alt="Delete Icon"
                      data-nimg="1"
                      decoding="async"
                      height="24"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                      style="color: transparent;"
                      width="24"
                    />
                  </button>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
